src/Traits/AuthTrait.php

```diff
@@ -22,6 +22,298 @@
+
+        if (empty($request)) {
+            $request = $this->request->getJsonRawBody(true);
+        }
+
+        $user->email = $this->request->getPost('email', 'email');
+        $user->firstname = ltrim(trim($this->request->getPost('firstname', 'string')));
+        $user->lastname = ltrim(trim($this->request->getPost('lastname', 'string')));
+        $user->password = ltrim(trim($this->request->getPost('password', 'string')));
+        $userIp = !defined('API_TESTS') ? $this->request->getClientAddress() : '127.0.0.1'; //help getting the client ip on scrutinizer :(
+        $user->displayname = ltrim(trim($this->request->getPost('displayname', 'string')));
+        $user->defaultCompanyName = ltrim(trim($this->request->getPost('default_company', 'string')));
+
+        //Ok let validate user password
+        $validation = new Validation();
+        $validation->add('password', new PresenceOf(['message' => _('The password is required.')]));
+        $validation->add('firstname', new PresenceOf(['message' => _('The firstname is required.')]));
+        $validation->add('email', new EmailValidator(['message' => _('The email is not valid.')]));
+
+        $validation->add(
+            'password',
+            new StringLength([
+                'min' => 8,
+                'messageMinimum' => _('Password is too short. Minimum 8 characters.'),
+            ])
+        );
+
+        $validation->add('password', new Confirmation([
+            'message' => _('Password and confirmation do not match.'),
+            'with' => 'verify_password',
+        ]));
+
+        //validate this form for password
+        $messages = $validation->validate($this->request->getPost());
+        if (count($messages)) {
+            foreach ($messages as $message) {
+                throw new Exception($message);
+            }
+        }
+
+        //user registration
+        try {
+            $this->db->begin();
+
+            Auth::signUp($request);
+
+            $this->db->commit();
+        } catch (Exception $e) {
+            $this->db->rollback();
+
+            throw new Exception($e->getMessage());
+        }
+
+        $token = $user->getToken();
+
+        //start session
+        $session = new Sessions();
+        $session->start($user, $token['sessionId'], $token['token'], $userIp, 1);
+
+        $authSession = [
+            'token' => $token['token'],
+            'time' => date('Y-m-d H:i:s'),
+            'expires' => date('Y-m-d H:i:s', time() + $this->config->jwt->payload->exp),
+            'id' => $user->getId(),
+        ];
+
+        $user->password = null;
+
+        return $this->response([
+            'user' => $user,
+            'session' => $authSession
+        ]);
+    }
+
+    /**
+     * Recover user information, by getting the email for the reset pass form.
+     *
+     * @method POST
+     * @url /v1/recover
+     *
+     * @return Response
+     */
+    public function recover() : Response
```

Comment:
@rwhite27 have to cehck with the function in the authcontroller, if we overwrite it that means 2 thing either we remove it from here or move the other ones here https://github.com/bakaphp/canvas-core/blob/0.2/src/Api/Controllers/AuthController.php#L349